### ISSUE TITLE
B3 results

### ISSUE THREAD

**Issue by jeffz0** [drimpossible/GDumb#4]
Hi, 

First off, congrats on publishing in ECCV, this is a very insightful paper questioning some of the recent works in incremental learning. My question is how did you achieve such high results in B3 (58.4% cifar100 and 62.86% ImageNet) if for every task, the model is trained from scratch with only 20 images per class. Isn't that too small of a number of examples to properly train resnet32 or resnet18? Is it because gdumb uses heavy data augmentation?

Thanks,

Jeff

**Comment by drimpossible** [comment]
Hi,

B3 was the only setup where we use the pretrained model (trained on 50 classes) as the starting point for training rather than entirely from scratch, following the setup in the original paper-- since our goal is to replace the continual part of training. We train that model using the same training process with 20 images per class. 

Hope this allows you to replicate the accuracy results.

**Comment by jeffz0** [comment]
Thank you for your response, that helps a lot! I also had a few question about A1 formulation. 

In Table 2, I notice for CIFAR10 you use Resnet-18.  Are you upsampling the 32x32 images to 224x224? Are you doing a single pass or multiple passes through the data?

I am also amazed with just 500 training images, you can train Resnet-18 from scratch to 45.8% on Cifar10. Looking through the code, it seems you only do random resize crop. Were there other data augmentation methods you used? Or is 50 images per class enough for training large resnet models from scratch?

Thanks again!

Jeff

**Comment by drimpossible** [comment]
`Do you upsample?`
No, we don't. The ResNet18 architecture for CIFAR is just a much fatter (channelwise) version of CIFAR-ResNets given https://github.com/drimpossible/GDumb/blob/master/src/models/cifar/resnet.py in this script.

Our formulation is online, yes -- in that sense we make only a single pass through the data but the network itself never sees any of the online samples. We make multiple passes over the memory similar to other online approaches.

`you can train Resnet-18 from scratch to 45.8% on Cifar10... is 50 images per class enough for training large resnet models from scratch?`
Yes, in A1 we do achieve those numbers from scratch with just 500 images. 50 images per class is indeed sufficient, which is what makes our paper surprising: the fact that even after throwing away all the online samples, you can achieve the same (or better) performance. 

`Looking through the code, it seems you only do random resize crop. Were there other data augmentation methods you used?`
We do not do randomresizedcrop for CIFAR, we use the augmentations for 32x32 inputs, which are these three:  `torchvision.transforms.RandomCrop(inp_sz, padding=4), torchvision.transforms.RandomHorizontalFlip()` (corrected:1/2)

Always happy to help! Please don't hesitate to ask questions.

-Ameya

**Comment by jeffz0** [comment]
These are very helpful answers, thank you. Just a final clarification on your resnet model. In the code right now, for cifar you only have depths 20, 32, 44, 56, 110, and 1202. Are you just changing the depth to 18 and making it fatter? 

If so, I've seen resnet-18 being used for cifar in other papers as well. Is this standard practice? or do most other papers use imagenet's resnet-18 and upsample cifar images to 224x224?

**Comment by drimpossible** [comment]
Ah! I see the 'nettype being imagenet' is what confused you. Sorry! The nettype is just for my reference, all networks in the cifar folder take 32x32 sized inputs-- just the networks are fatter. 

Yes, using this huge ResNet18/50 architecture with 32x32 inputs is standard practice. I think Resnet20 would simply be vastly more efficient but I am not sure why people do this, hence I might be missing something important.

**Comment by jeffz0** [comment]
Ok, makes sense. I greatly appreciate all the clarifications!
